vcf2fhir/fhir_helper.py

```diff
@@ -350,40 +421,89 @@ def add_variant_obv(self, record, ref_seq, ratio_ad_dp):
                         ]
                     }
                 )
-            observation_dv.component.append(observation_dv_component3)
+            observation_dv.component.append(observation_dv_component4)
 
-        observation_dv_component4 = observation.ObservationComponent()
-        observation_dv_component4.code = concept.CodeableConcept(
-            {
-                "coding": [
+        if ((not record.is_sv) and alleles['FREQUENCY'] != ""):
+            observation_dv_component5 = observation.ObservationComponent()
+            observation_dv_component5.code = concept.CodeableConcept(
+                {
+                    "coding": [
+                        {
+                            "system": "http://loinc.org",
+                            "code": "81258-6",
+                            "display": "Sample VAF"
+                        }
+                    ]
+                }
+            )
+            observation_dv_component5\
+                .valueQuantity = quantity.Quantity(
                     {
-                        "system": "http://loinc.org",
-                        "code": "69547-8",
-                        "display": "Genomic Ref allele [ID]"
+                        "system": "http://unitsofmeasure.org",
+                        "code": "1",
+                        "value": float(alleles['FREQUENCY'])
                     }
-                ]
-            }
-        )
-        observation_dv_component4.valueString = record.REF
-        observation_dv.component.append(observation_dv_component4)
+                )
+            observation_dv.component.append(observation_dv_component5)
 
-        observation_dv_component5 = observation.ObservationComponent()
-        observation_dv_component5.code = concept.CodeableConcept(
+        if record.is_sv and record.INFO['SVTYPE'] in ['CNV', 'DUP', 'DEL']:

```

Comment:
I see the each if has `record.is_sv ` or `not `record.is_sv` why not create a nested if and remove the repeated condition from each `if`.
```
if  record.is_sv:
   // explain contion 1. Provide some example If logic is complicated.(Repeat same thing for all conditions.)
    if condition_1:
    if condition_2:
else:
    if condition_3:
    if condition_4:
```
Conditional codes are complicated to understand and very error-prone. You should always try to simplify the conditions as much as possible and try to make them more readable. Unit tests are also very important to make sure all the conditions work as expected.